packages/ui/src/forum/modals/PostHistoryModal/PostHistoryModal.tsx

```diff
@@ -24,7 +25,23 @@ export const PostHistoryModal = React.memo(() => {
     modalData: { postId, author },
   } = useModal<PostHistoryModalCall>()
   const { isLoading, edits } = useForumPostEdits(postId)
-  const [activeEdit] = useState(0)
+  const [activeEdit, setActiveEdit] = useState(0)
+
+  const editsInView = useMemo(() => new Array<boolean>(edits?.length ?? 0), [edits?.length])
+
+  const pickActiveEdit = useCallback(
+    (index: number) => (isCurrentInView: boolean) => {
+      editsInView[index] = isCurrentInView
+      const newActiveEdit = editsInView.reduceRight(
+        (currentTopIndex, isEditInView, index) => (isEditInView ? index : currentTopIndex),
+        0
```

Comment:
As we're simply looking for the first occurrence of `true`, this can be simplified to:
```suggestion
      const newActiveEdit = editsInView.findIndex((value => value))
```